### DISCUSSION TITLE
Allow Scripts

### DISCUSSION THREAD

**Discussion by Rand0mRange** [cefsharp/CefSharp#4324] [General]
Hi all.
How to allow script execution in iframes?

The site creates multiple Iframes, I use EvaluateScriptAsync to inject the script, but the site on child Iframes forbids executing the injected script. Is it possible to enable a flag of some kind in the browser (--allow-scripts ), or is there another solution? This is the error coming from the site.

Blocked script execution in 'about:blank' because the document's frame is sandboxed and the 'allow-scripts' permission is not set.

**Comment by amaitland** [comment]
In future please use [Discussions](https://github.com/cefsharp/CefSharp/discussions) to ask support related questions.

>  I use EvaluateScriptAsync to inject the script

How are you executing the script exactly? Are you executing specifically against the frame or are you trying to access the frame via javascript? Please provide a code example.

**Reply by Rand0mRange** [reply]
If done through FrameLoadStart, then the script will be implemented only for the main page. In my case, the main page has several more IFrames , and in order for my scripts to be injected for all frames, I use RenderProcessMessageHandler. Scripts are embedded in each frame, everything is fine here, but there is a frame in which Canvas is read, the function of which I redefine accordingly. And as soon as my script execution happens I get this error - Blocked script execution in 'about:blank' because the document's frame is sandboxed and the 'allow-scripts' permission is not set. Because this Iframe has the sandbox attribute set and the Allow Scripts attribute is not set in order for my script to execute. I'm willing to pay money if you help solve this issue.

I can implement the necessary scripts, I need to get access to the ability to execute a script inside an IFrame that has the sandbox attribute set and the required attribute to run the Allow-Scripts script is not set - https://browserleaks.com/canvas

![image](https://user-images.githubusercontent.com/11533561/206136777-9168af3c-428d-44e9-80ab-cd5ffd49a42b.png)

**Reply by amaitland** [reply]
> If done through FrameLoadStart, then the script will be implemented only for the main page

That's not correct. You can execute scripts against the individual frame, you cannot access the DOM at this point though.

Without a code example I cannot provide any additional information.
